pkg/crd/testdata/testdata.kubebuilder.io_cronjobs.yaml

```diff
@@ -7340,9 +7341,11 @@ spec:
                 minLength: 4
                 type: string
               unprunedEmbeddedResource:
+                allOf:
```

Comment:
This looks strange, not sure where I should look at.

Reply:
@joelanford @alvaroaleman do you have an idea where I should look in the code to find why this gets transformed into `allOf` ?

Reply:
From what I understand it happens in the flattener. I’m a bit lost in the code though. Any hint would be appreciated.

Reply:
I think it happens when `kubebuilder:pruning:PreserveUnknownFields` is present on a field and the target type. I can reproduce with any struct. Is this a bug ? Should both markers get merged when generating the CRD ? cc @alvaroaleman @joelanford